src/api/config/routes/api.rb

```diff
@@ -346,6 +346,12 @@
   end
 end
 
+scope :labels do
+  resources :projects, only: [], param: :name, constraints: cons do
+    resources :labels, only: %i[index create destroy], path: '', controller: 'labels/projects', param: :id, constraints: cons
```

Comment:
```suggestion
  resources :projects, only: [], param: :name do
    resources :labels, only: %i[index create destroy], path: '', constraints: cons
```
We don't need the constraints in the `resources :projects ...` line. That line doesn't add any route.

We could simplify the resources :labels ... line. That is what the defaults do.

Reply:
`resources :projects` adds `/projects/:project_name` to the route, with a param we need to constraint

Reply:
As for the labels line:

```diff
-    project_labels GET          /labels/projects/:project_name(.:format)            labels/projects#index {project_name: /[^\/]*/}
-                   POST         /labels/projects/:project_name(.:format)            labels/projects#create {project_name: /[^\/]*/}
-     project_label DELETE       /labels/projects/:project_name/:id(.:format)        labels/projects#destroy {id: /\d*/, project_name: /[^\/]*/}
+    project_labels GET          /labels/projects/:project_name(.:format)            labels#index {project_name: /[^\/]*/}
+                   POST         /labels/projects/:project_name(.:format)            labels#create {project_name: /[^\/]*/}
+     project_label DELETE       /labels/projects/:project_name/:id(.:format)        labels#destroy {id: /\d*/, project_name: /[^\/]*/}
```

Reply:
> `resources :projects` adds `/projects/:project_name` to the route, with a param we need to constraint

Sure. It adds `/projects/:project_name` to the route. But it doesn't add any route.

That's taken in account in the `resources :labels ...` line.

Reply:
> As for the labels line:
> 
> ```diff
> -    project_labels GET          /labels/projects/:project_name(.:format)            labels/projects#index {project_name: /[^\/]*/}
> -                   POST         /labels/projects/:project_name(.:format)            labels/projects#create {project_name: /[^\/]*/}
> -     project_label DELETE       /labels/projects/:project_name/:id(.:format)        labels/projects#destroy {id: /\d*/, project_name: /[^\/]*/}
> +    project_labels GET          /labels/projects/:project_name(.:format)            labels#index {project_name: /[^\/]*/}
> +                   POST         /labels/projects/:project_name(.:format)            labels#create {project_name: /[^\/]*/}
> +     project_label DELETE       /labels/projects/:project_name/:id(.:format)        labels#destroy {id: /\d*/, project_name: /[^\/]*/}
> ```

Ops... true when removing `controller: 'labels/projects'`.

However, we can still remove `param: :id`.

Reply:
That I did, thanks!